src/App.tsx

```diff
@@ -1,87 +1,96 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 import { remult } from "remult";
 import { Task } from "./shared/Task";
 import { TasksController } from "./shared/TasksController";
+import { observer } from "mobx-react-lite";
+import { action, makeAutoObservable, observable, runInAction } from 'mobx';
 
-const taskRepo = remult.repo(Task);
 
-async function fetchTasks(hideCompleted: boolean) {
-  return taskRepo.find({
-    limit: 20,
-    orderBy: { completed: "asc" },
-    where: { completed: hideCompleted ? false : undefined }
-  });
-}
 
 
+class Store {
+  taskRepo = remult.repo(Task);
+  readonly tasks = observable<Task>([]);
+  hideCompleted = false;
+  constructor() {
+    makeAutoObservable(this);
+  }
+  replaceTasks(t: Task[]) {
+    this.tasks.replace(t.map(t => makeAutoObservable({ ...t })));
```

Comment:
Won't `this.tasks = t.map(t => makeAutoObservale({...t}))` do the same?
Also, "t"? I wouldn't abbreviate.

Reply:
I thought that `readonly tasks = observable<Task>([]);` is a best practice  - but from your feedback - it seems that it's not.

👍